lib/paper_trail/model_config.rb

```diff
@@ -141,17 +149,24 @@ def cannot_record_after_destroy?
     def setup_versions_association(klass)
       klass.has_many(
         klass.versions_association_name,
-        lambda do |object|
+        lambda do
           relation = order(model.timestamp_sort_order)
-          item_type = object.paper_trail.versions_association_item_type
-          unless item_type.nil?
-            relation = relation.unscope(where: :item_type).where(item_type: item_type)
-          end
+          item_type = klass.paper_trail.send(:versions_association_item_type)
+          relation = relation.unscope(where: :item_type).where(item_type: item_type)
           relation
         end,
         class_name: klass.version_class_name,
         as: :item
       )
+
+      # We override the assocation when STI models are created from a base class
+      # in order to use the right `item_type`.
+      klass.singleton_class.prepend(Module.new {
+        def inherited(klass)
+          super
+          paper_trail.send(:setup_versions_association, klass)
+        end
+      })
```

Comment:
Love this idea!  Worked out well for the fixes for CelebrityFamily tests and migrations to work.